src/syscheckd/fim_diff_changes.c

```diff
@@ -405,8 +406,9 @@ char *fim_file_diff(const char *filename) {
 
     // Generate diff structure
     diff_data *diff = initialize_file_diff_data(filename);
-    if (!diff){
-        goto cleanup;
+    if (!strcmp("error", diff->compress_file)){
```

Comment:
The function `initialize_file_diff_data()` should return null in case of error

Reply:
Done